### DISCUSSION TITLE
How do I build a React library with this?

### DISCUSSION THREAD

**Discussion by alvarlagerlof** [swc-project/swc#1998] [Q&A] [answered]
I want to build a npm package that exports a react component and some hooks. Building works fine, but I keep on getting this 
`Module not found: Can't resolve 'react'` when a next site imports it. I have marked react and react-dom as a peer decency too.

Here's my config:
```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": true,
      "decorators": true,
      "dynamicImport": true
    },
    "transform": {
      "legacyDecorator": true,
      "decoratorMetadata": true
    },
    "target": "es2018",
    "keepClassNames": true,
    "loose": true
  },
  "module": {
    "type": "commonjs",
    "strict": true,
    "strictMode": true,
    "lazy": false,
    "noInterop": true
  },
  "sourceMaps": "inline"
}
```

**Comment by alvarlagerlof** [comment, marked as answer]
My issue wasn't swc but the fact that I hadn't included react and react-dom in devDependencies

**Reply by kdy1** [reply]
Did you install react from the project which uses your library?

**Reply by alvarlagerlof** [reply]
Yes. Well next.js does.
